src/common/types.cpp

```diff
@@ -434,7 +434,15 @@ LogicalType TransformStringToLogicalType(const string &str) {
 
 LogicalType GetUserTypeRecursive(const LogicalType &type, ClientContext &context) {
 	if (type.id() == LogicalTypeId::USER && type.HasAlias()) {
-		return Catalog::GetSystemCatalog(context).GetType(context, SYSTEM_CATALOG, DEFAULT_SCHEMA, type.GetAlias());
+	}
+	if (type.id() == LogicalTypeId::USER && type.HasAlias()) {
+		auto catalog_type = Catalog::GetSystemCatalog(context).GetType(context, DEFAULT_SCHEMA, type.GetAlias(),
```

Comment:
This seems to be manually doing catalog resolution - I think this can be replaced/simplified with `Catalog::GetType(context, INVALID_CATALOG, INVALID_SCHEMA, type.GetAlias())`

Reply:
You are absolutely right! Thanks for bringing it up!